### PR TITLE
add entt/3.3.0

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **entt/3.3.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

closes https://github.com/conan-io/conan-center-index/issues/941

### CODE REVIEW

File: recipes/entt/3.x.x/conanfile.py [hunk 1]
```diff
@@ -4,19 +4,17 @@
...
     topics = ("conan," "entt", "gaming", "entity", "ecs")
-
     homepage = "https://github.com/skypjack/entt"
     url = "https://github.com/conan-io/conan-center-index"
-
     license = "MIT"
-
     settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake", "cmake_find_package_multi"
+    generators = "cmake"
```
Comment (madebr) [inline recipes/entt/3.x.x/conanfile.py]: This recipe has no own `CMakeLists.txt` and does not call `conan_basic_setup()` anywhere.

Comment (SpaceIm) [reply recipes/entt/3.x.x/conanfile.py]: ok, generators removed, but also most of settings values (except compiler), and added no_copy_source

File: recipes/entt/3.x.x/conanfile.py [hunk 2]
```diff
@@ -53,26 +51,28 @@ def configure(self):
...
+        self._cmake.definitions["BUILD_TESTING"] = False
+        self._cmake.definitions["BUILD_DOCS"] = False
+        self._cmake.configure(
             source_folder=self._source_subfolder,
             build_folder=self._build_subfolder
         )
-        return cmake
+        return self._cmake
 
     def build(self):
```
Comment (danimtb) [inline recipes/entt/3.x.x/conanfile.py]: If this is a header-only, why do we need the build step?

Comment (SpaceIm) [reply recipes/entt/3.x.x/conanfile.py]: It has targets. Kitten recipe does the same thing (and maybe the c++17 in Kitten recipe could be used in entt recipe).
Honestly, we could just copy src dir and don't even care about CMake.

Comment (danimtb) [reply recipes/entt/3.x.x/conanfile.py]: ok, we will take care of that in another PR. Thanks!

**Pull request merged**